### ISSUE TITLE
grid lines not rendered correctly

### ISSUE THREAD

**Issue by isaacwhanson** [godotengine/godot#6003]
**Operating system or device - Godot version:**
Linux x86_64 - Godot_v2.1_rc1_x11.64

**Issue description** (what happened, and what was expected):
![screenshot](http://i.imgur.com/rvl6Dkz.png)
_Bugsquad edit: inlined screenshot as image_
the red/green/blue lines that represent the axes and grid lines do not properly intersect (far from it).
this is a new, empty project.

**Steps to reproduce:**
Download the binary version of Godot 2.1 x11 x64.  run it and create a new project.
Rotate the view so it is no longer axis-aligned.  Persists after closing and reopening Godot.

**Link to minimal example project** (optional but very welcome):
truly just a new empty project.

**Some hardware info:**
`lspci | grep -e VGA -e 3D`
00:02.0 VGA compatible controller: Intel Corporation 4th Gen Core Processor Integrated Graphics Controller (rev 06)
02:00.0 3D controller: NVIDIA Corporation GK107GLM [Quadro K1100M](rev ff)

**Comment by bojidar-bg** [comment]
Can't confirm on my end, though I'm on the same CPU arch. Seems driver related?

@isaacwhanson can you try updating drivers, etc.?

**Comment by akien-mga** [comment]
I have the same issue with Intel HD Graphics 4000, fully up-to-date Linux drivers (mesa 12.0.1, latest xorg, corresponding intel driver from git).

**Comment by akien-mga** [comment]
IIRC @reduz told me that it's a drivers/GL 2.1 issue and there's not much that can be done about it. Might get better with the GLES 3.0 backend for Godot 3.0 :)

**Comment by hurikhan** [comment]
Looks like it is only releated to the intel driver.
I can also reproduce this on the integrated intel gpu on linux.

When I start godot on the dedicated gpu (radeon_si) with:
```
xrandr --setprovideroffloadsink 1 0
DRI_PRIME=1 ./bin/godot.x11.tools.64.llvm
```
The grid looks fine.

**Comment by JulienGio** [comment]
Can't reproduce with Nvidia GPU, so yeah...

**Comment by isaacwhanson** [comment]
I have a dedicated nvidia GPU setup with bumblebee/primus in addition to the intel integrated GPU.  If I offload to the dedicated nvidia it works fine.
`primusrun ./Godot_v2.1_rc1_x11.64`

**Comment by bojidar-bg** [comment]
And BTW, this bug is also present in blender for me... :confused:

**Comment by 648trindade** [comment]
Same for:
Debian Buster/Testing 64 bits up-to-date
Godot versions < 3 (not present on v3 alpha)
CPU: Intel i5 4210u
GPU: Intel HD 4400 (Haswell) (i915 linux kernel driver)

I've opened an issue cause i couldn't find this issue, now it is closed:
https://github.com/godotengine/godot/issues/10214

**Comment by Calinou** [comment]
This bug does not seem to occur anymore as of Godot 3.0 (as mentioned by @648trindade), so I'm closing this. Feel free to open a new issue with a screenshot and detailed hardware information in case you can still experience this bug.

**Comment by jv110** [comment]
Isn't 2.1 still being maintained? The fix should be brought over to 2.1.5 (or maybe 2.1.6).

**Comment by akien-mga** [comment]
The fix was to rewrite the renderer, I don't think we'll do that for 2.1 ;)
